File: TypeScript/TrapezoidalRule.ts
import * as readline from 'readline'

const readInterface = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

const f =(x: number)=>{
    return x*x + 5.0 * x + 6.0
}

const TrapezoidalRule = (valueA:number, valueB:number) =>{

    const N = 10000

    const dx = (valueB - valueA)/N
    let s = (f(valueA) + f(valueB)) / 2.0

    for (let i = 1; i <= N -1; i++){
        s += f(valueA + dx * i)
    }

    s *= dx

    return s
}


readInterface.question('定積分する区間Aを入力してください',(n:string)=>{

    const valueA = Number(n)

    readInterface.question('定積分する区間Bを入力してください',(n:string)=>{
        readInterface.close()
    
        const valueB = Number(n)

        try{
            const result = TrapezoidalRule(valueA, valueB)
            console.log(`近似値は${result}です。`)
        }
        catch(e){
            console.log(e)
        }
    })
})

File: Python/trapezoidal_rule.py

def f(x: float)-> float:
    return x**2 + 5.0 * x + 6.0


def trapezoidal_rule (value_a:float, value_b:float)->float:

    """
    台形法により、関数f(x)= x^2+5x+6を区間[a, b]で定積分した近似値を求める


    Parameters
    ----------
    value_a: float
        区間a
    valie_b: float
        区間b

    Returns
    -------
    result: float
        入力された区間で定積分した近似値
    """

    if type(value_a) is not float or type(value_b) is not float:
        raise TypeError('入力された整数がfloat型ではありません')

    N = 10000

    dx = (value_b - value_a) / N
    s = (f(value_a) + f(value_b)) / 2.0

    for i in range(1, N - 1):
        s = s + f(value_a + dx * i)
    s = s * dx

    return s


if __name__ == '__main__':

    try:
        value_a , value_b = (float(x) for x in input('区間[a, b]を入力してください:').split())
        # value_a = 'a'
        # value_b = 'b'

        result = trapezoidal_rule(value_a, value_b)
        
        print('近似値は{:.08f}です。'.format(result))
        
    except TypeError as e:
        print(e)
